Sources/Core/Presenter/Channel/ChannelPresenter+Events.swift

```diff
@@ -161,8 +161,13 @@ extension ChannelPresenter {
     }
     
     private func appendOrUpdateMessageItem(_ message: Message, at index: Int = -1) {
-        if let lastMessage = lastMessage, (message.created > lastMessage.created || message.id == lastMessage.id) {
+        switch lastMessage {
+        case .none:
             lastMessageAtomic.set(message)
+        case .some(let lastMessage):
+            if message.created > lastMessage.created || message.id == lastMessage.id {
+                lastMessageAtomic.set(message)
+            }
         }
```

Comment:
Thanks for your help! 
I first time see to use optional with switch 😀

Why you didn't use `||` in `if`?
```suggestion
        if lastMessage == nil || message.created > lastMessage.created || message.id == lastMessage.id {
            lastMessageAtomic.set(message)
        }
```
I this will work too, because the `message` is not optional and we can omit unwrapping.

Reply:
@buh In your solution it'll have to be force unwrapped:
```swift
if lastMessage == nil || message.created > lastMessage!.created || message.id == lastMessage!.id {
      lastMessageAtomic.set(message)
}
```
so I think current solution is better.

Reply:
I tested my solution on the wrong target and it was compiled.